packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js

```diff
@@ -156,6 +156,64 @@ export function regexEncode(value) {
   return value.replace(/[-[\]/{}()*+?.\\^$|]/g, '\\$&');
 }
 
+export function parseXML(string) {
+  return new window.DOMParser().parseFromString(string, 'text/xml');
+}
+
+export class TimeSpan {
+  constructor(time, isMilliseconds = true) {
+    if (isMilliseconds) {
+      this.milliseconds = time;
+    } else {
+      this.milliseconds = time * 1000;
+    }
+  }
+
+  getMilliseconds() {
+    return this.milliseconds;
+  }
+
+  getSeconds() {
+    return this.getMilliseconds() / 1000;
+  }
+
+  getMinutes() {
+    return this.getSeconds() / 60;
+  }
+
+  getHours() {
+    return this.getMinutes() / 60;
+  }
+
+  getDays() {
```

Comment:
Is that `TimeSpan` expected to handle short duration like the length of a video, or long duration like a document's age?

If we're targeting only short duration, I would remove `getDays` and `getWeeks`. If we're targeting long duration, I would also add `getMonths` and `getYears`. But I'm wondering if we should use a lib like [momentjs](https://momentjs.com/) to handle relative time.

Reply:
Right now it's only used for youtube video durations but I figured why not add the whole thing so it'll be there if we (or our users) need it.
I copied this util class from the JSUI to avoid myself the hassle. A node lib like momentjs could be good. However my reticence over adding external deps is that right now our static ressources only contain headless and I think that's nice. We might need to add bueno for extra validation utils in the future but I'm wary of adding entire libs if our needs can be met by such a simple class like this.